subsys/net/openthread/rpc/common/ot_rpc_common.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "ot_rpc_common.h"
+
+#include <nrf_rpc/nrf_rpc_serialize.h>
+
+#include <openthread/dataset.h>
+
+void ot_rpc_decode_error(const struct nrf_rpc_group *group, struct nrf_rpc_cbor_ctx *ctx,
+			 void *handler_data)
+{
+	nrf_rpc_rsp_decode_i32(group, ctx, handler_data);
+}
+
+void ot_rpc_decode_void(const struct nrf_rpc_group *group, struct nrf_rpc_cbor_ctx *ctx,
+			void *handler_data)
+{
+	nrf_rpc_rsp_decode_void(group, ctx, handler_data);
+}
+
+void ot_rpc_decode_dataset_tlvs(const struct nrf_rpc_group *group, struct nrf_rpc_cbor_ctx *ctx,
+				void *handler_data)
+{
+	otOperationalDatasetTlvs *dataset = *(otOperationalDatasetTlvs **)handler_data;
+	const void *data;
+	size_t length;
+
+	data = nrf_rpc_decode_buffer_ptr_and_size(ctx, &length);
+
+	if (data == NULL) {
+		handler_data = NULL;
```

Comment:
This does nothing. You should probably make `dataset` of type `otOperationalDatasetTlvs *` and then do `*dataset = NULL` in here.

Reply:
Didn't get your statement.
It does what is written here, writes NULL to notify the function above about absence of data. `dataset` of type `otOperationalDatasetTlvs *` has been made a couple of lines above.
Please clarify what you meant.

Reply:
`handler_data` is a local variable,  so you're not updating the user provided variable anyhow.

Reply:
You're right, good catch. Will fix it.